packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  updateAuthAddUserGroups,
+  amplifyPushWithoutCodegen,
+  addGeofenceCollectionWithDefault,
+  populateGeofencesWithDefault,
+} from 'amplify-e2e-core';
+import { existsSync } from 'fs';
+import path from 'path';
+
+describe('amplify geo populate', () => {
+  let projRoot: string;
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('geo-populate-test');
+  });
+
+  afterEach(async () => {
+    const metaFilePath = path.join(projRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+    if (existsSync(metaFilePath)) {
+      await deleteProject(projRoot);
+    }
+    deleteProjectDir(projRoot);
+  });
+
+  it('should throw error if there is no geofence collection provisioned', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot);
+    await expect(populateGeofencesWithDefault(projRoot)).rejects.toThrowError();
```

Comment:
Can we move the steps from `initJS...` until `addGeofenceCollection...` to the test setup as we are only testing the populate workflow in all the tests.

Reply:
I am considering compress the similar tests into one case where we provision once and upload multiple geo json files, which will save much time on total run time

Reply:
That might make debugging difficult in case something fails. I would say as a best practice test one specific case in a test.

Reply:
Update: The cases are still separate but only has one provision run